src/core/datatypes/trees/TopologyNode.cpp

```diff
@@ -721,6 +721,73 @@ bool TopologyNode::equals(const TopologyNode& node) const
     return true;
 }
 
+RbBitSet TopologyNode::fillCladeBitset(std::map<RbBitSet, TopologyNode*> &clade_index_map, size_t num_tips, bool include_tips)
+{
+    RbBitSet this_bs = RbBitSet(num_tips);
+    if ( isTip() == true )
+    {
+        // We can't use indices for tree comparison because different trees may
+        // have different indices for the same taxon.
+        // Instead make the BitSet ordered by taxon names.
+        // Eventually this should be refactored with the TaxonMap class.
```

Comment:
I was working on probably the same problem from another direction -- see for example #294.  Instead of doing this inside TopologyNode, we ensure that all the trees in a trace use the same TaxonMap.  This ensures that different trees DO have the same indices for the same taxon.  The attempt to force a tree to use a given TaxonMap also finds problems if the tree is missing a taxon in the TaxonMap.